src/compiler/checker.ts

```diff
@@ -18660,7 +18660,13 @@ namespace ts {
                     (augmentations || (augmentations = [])).push(file.moduleAugmentations);
                 }
                 if (file.symbol && file.symbol.globalExports) {
-                    mergeSymbolTable(globals, file.symbol.globalExports);
+                    // Merge in UMD exports with first-in-wins semantics (see #9771)
+                    const source = file.symbol.globalExports;
+                    for (const id in source) {
+                        if (hasProperty(source, id) && !hasProperty(globals, id)) {

```

Comment:
I don't think you need to check `hasProperty(source, id)` since we switch to use new "Map"?